.github/workflows/ci-pipeline.yaml

```diff
@@ -63,136 +35,67 @@ jobs:
+          if [[ "${#flavors[@]}" -le "0" ]]; then
+            echo "No changes detected in containers. The rest of the steps will be skipped."
             echo "::set-output name=result::skip"
+          else
+            containers_json=$(printf "%s\n" "${flavors[@]}" | jq -R . | jq -cs .)
+            echo "::set-output name=result::ok"
+            echo "::set-output name=containers::${containers_json}"
           fi
-      # Using actions/github-scripts because using exit 1 in the script above would not provide any output
-      # Source: https://github.community/t/no-output-on-process-completed-with-exit-code-1/123821/3
-      - id: show-error
-        name: Show error
-        if: ${{ steps.get-container.outputs.result == 'fail' }}
-        uses: actions/github-script@v3
-        with:
-          script: |
-              core.setFailed('${{ steps.get-container.outputs.error }}')
   vib-verify:
     runs-on: ubuntu-latest
-    needs: get-container
-    if: ${{ needs.get-container.outputs.result == 'ok' && contains(github.event.pull_request.labels.*.name, 'verify') }}
-    name: Verify
+    needs: get-containers
+    if: ${{ needs.get-containers.outputs.result == 'ok' && contains(github.event.pull_request.labels.*.name, 'verify') }}
+    name: Verify container
+    continue-on-error: false
+    strategy:
+      fail-fast: false
+      matrix:
+        container: ${{ fromJSON(needs.get-containers.outputs.containers) }}
     steps:
       - uses: actions/checkout@v3
         name: Checkout Repository
         with:
-          # Required to search the latest commit with the tag
-          fetch-depth: 0
+          # Full history is not required anymore
+          fetch-depth: 1
           # labeled events trigger the event with the latest commit in main
           ref: ${{ github.event.pull_request.head.ref }}
           repository: ${{ github.event.pull_request.head.repo.full_name }}
-      - id: get-tag
-        name: Get latest image tag
+      - id: get-container-metadata
+        name: Get image tag and container name
         run: |
-          # Now EAM commits has the following format "[bitnami/<container>] Release <tag>"
-          tag="$(git log --pretty=tformat:"%s" -n 1 --grep="[R|r]elease" --author bitnami-bot@vmware.com --author containers@bitnami.com --author containers-bot@bitnami.com -- ${{ needs.get-container.outputs.path }} | sed "s|\[.*\]||" | sed "s|[R|r]elease||" | awk '{print $1}')"
-          if [[ -z "${tag}" ]]; then
-            echo "No tag found for: ${{ needs.get-container.outputs.path }}"
-            exit 1
-          else
+          if [[ -d "${{ matrix.container }}" ]]; then
+            name="$(echo "${{ matrix.container }}" | awk -F '/' '{print $2}')"
+            # Build a tag based on current SHA
+            tag="$(echo "${{ matrix.container }}" | awk -F '/' -v run_number="${GITHUB_RUN_NUMBER}" '{printf "%s-rc.%s", $3, run_number}')"
```

Comment:
Why can't we retrieve the tag from the Dockerfile label as we do in the CD and instead create our custom one?

Reply:
Because at the moment is not mandatory for the users to increase that label in the Dockerfile and it could overlaps with an existing version.